nomos-services/utils/src/overwatch/status/service_status_entry/error.rs

```diff
@@ -0,0 +1,40 @@
+//! Provides conversion from a collection of [`ServiceStatusEntry`] to
+//! the general-purpose [`DynError`](overwatch::DynError) type.
+
+use std::fmt::{Debug, Display, Formatter};
+
+use crate::overwatch::status::ServiceStatusEntry;
+
+#[derive(Debug)]
+pub struct ServiceStatusEntriesError<RuntimeServiceId: Display> {
+    service_status_entries: Vec<ServiceStatusEntry<RuntimeServiceId>>,
+}
+
+impl<RuntimeServiceId: Display> Display for ServiceStatusEntriesError<RuntimeServiceId> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let entries: Vec<String> = self
+            .service_status_entries
+            .iter()
+            .map(ToString::to_string)
+            .collect();
+        let entries = entries.join(", ");
+        write!(f, "ServiceStatuses: {entries}")
+    }
+}
+
+impl<RuntimeServiceId: Display + Debug> std::error::Error
+    for ServiceStatusEntriesError<RuntimeServiceId>
+{
+}
+
+impl<RuntimeServiceId, Iter> From<Iter> for ServiceStatusEntriesError<RuntimeServiceId>
+where
+    RuntimeServiceId: Display,
```

Comment:
If we remove the bound from the struct, we can remove it from here, since I don't see how implementing `Display` is useful in this context (?)

Reply:
Same as above, I believe.